src/main/java/org/jabref/gui/openoffice/OpenOfficePanel.java

```diff
@@ -511,12 +511,6 @@ private void pushEntries(CitationType citationType, boolean addPageInfo) {
         }
 
         final BibDatabaseContext bibDatabaseContext = stateManager.getActiveDatabase().get();
-        if (bibDatabaseContext == null) {
```

Comment:
Future: It is ok to add `assert` statements.

Example:

    assert bibDatabaseContext  != null

OK, one needs to add `-ea` somehow: https://stackoverflow.com/a/18168305/873282 -- I would have bet that I saw assertion errors when developing...

The long discussion at https://stackoverflow.com/q/2758224/873282 says: do something else, but I like assertions:

1. More semantic meaning than removed code
2. More semantic meaning than code comments
3. Runs in debug mode only (in contrast to `Objects.requireNonNull`
4. Does not clutter the code with `@NonNull` annotations, which need a separate framework for checking - (https://github.com/kelloggm/checkerframework-gradle-plugin)

Reply:
Okay, will follow that.
Reason for removal here was that `bibDatabaseContext` could never be null here.

Reply:
Sure. At the time of the code. But what if there is a future refactoring. One could argue that the whole research of pre- and post conditions is useless, because the author was sure at when writing the code 🤣🤣

This kind of code (or [JSpecify annitations](https://jspecify.dev/) are useful for working on the code at a later point of time. To avoid regressions.

Side track: Another tool for avoiding regressions is OpenFastTrace - see https://devdocs.jabref.org/requirements/ for details ^^.

Reply:
If it is to ensure the precondition, it is better to revert this change and use the dedicated error dialog instead of using an internal assertion. Will be updated in latest commit.

Reply:
Checks for empty database and databasecontext merged in (updated) lines 509-515.